core/src/tile/tileBuilder.cpp

```diff
@@ -74,7 +73,11 @@ std::shared_ptr<Tile> TileBuilder::build(TileID _tileID, const TileData& _tileDa
         builder.second->addLayoutItems(m_labelLayout);
     }
 
-    m_labelLayout.process();
+    ViewState viewState = m_scene->viewState();
+    viewState.zoom = _tileID.z;
```

Comment:
It still doesn't, the view state is always set to be based on the tile, thus explicit set of zoom and fractional zoom.